### PR TITLE
Add Torch ORT Callback, re-write callback section

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Adds the Torch ORT Callback to lightning bolts, refactor the docs to neaten the structure up + add helpers for tests.

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? \[not needed for typos/docs\]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/callbacks/torch_ort.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+        ...
+
+
+    model = MyTransformerModel()
+    trainer = Trainer(gpus=1, callbacks=ORTCallback())
+    trainer.fit(model)
+
+
+For even easier setup and integration, have a look at our Lightning Flash integration for `Text Classification <https://lightning-flash.readthedocs.io/en/latest/reference/text_classification.html#accelerate-training-inference-with-torch-ort>`__, `Summarization <https://lightning-flash.readthedocs.io/en/latest/reference/summarization.html#accelerate-training-inference-with-torch-ort>`__ and `Translation <https://lightning-flash.readthedocs.io/en/latest/reference/translation.html#accelerate-training-inference-with-torch-ort>`__.
```
Comment (ethanwharris) [inline docs/source/callbacks/torch_ort.rst:35]: We could maybe use intersphinx here to enable the link even if the heading is changed, would need to add a label above the required section in the flash docs for it to work

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Review by awaelchli — APPROVED** [review]

**Review by ananyahjha93 — APPROVED** [review]

**Review by ananyahjha93 — APPROVED** [review]

**Review by Borda — APPROVED** [review]
